libcontainer/container_linux.go

```diff
@@ -1017,7 +1017,14 @@ func (c *linuxContainer) Checkpoint(criuOpts *CriuOpts) error {
 
 	fcg := c.cgroupManager.GetPaths()["freezer"]
 	if fcg != "" {
-		rpcOpts.FreezeCgroup = proto.String(fcg)
+		err := c.checkCriuVersion(31400)
+		// CRIU currently cannot handle the v2 freezer correctly
+		// before release 3.14. For older releases we are telling
+		// CRIU to not use the cgroup v2 freezer. CRIU will pause
+		// each process manually using ptrace().
+		if err == nil || !cgroups.IsCgroup2UnifiedMode() {
+			rpcOpts.FreezeCgroup = proto.String(fcg)
```

Comment:
Can you move the `fcg := c.cgroupManager.GetPaths()["freezer"]` to under `if`, i.e. just above this line?

Reply:
i.e. something like
```go
if !cgroups.IsCgroup2UnifiedMode() &&  c.checkCriuVersion(31400) == nil {
  if fcg := c.cgroupManager.GetPaths()["freezer"]; fcg != "" {
    rpcOpts.FreezeCgroup = proto.String(fcg)
  }
}

My primary concern here is not call GetPaths() which is obsoleted for cgroup v2 unified mode

Reply:
> My primary concern here is not call GetPaths() in Cgroup2UnifiedMode case.

What I mean, in unified mode one should call `GetUnifiedPath()`.

Reply:
> What I mean, in unified mode one should call GetUnifiedPath().

Alas, this was not addressed and I overlooked it. Will followup with a PR